doc/design/libsvm_format_column.md

```diff
@@ -0,0 +1,134 @@
+# Support LibSVM Format Column in SQLFlow
+
+## Definition
+
+### LibSVM Format File
+
+A **LibSVM format file** is a text file storing sparse data, in which each line is like:
+
+```
+<label> <index>:<value> <index>:<value> ...
+```
+
+where:
+
+- `<label>` is the label value of the sample and it is optional. For example, `<label>` is optional when predicting. 
+- `<index>:<value>` indicates each non-zero index-value pair in the sparse data. There may be many index-value pairs in each line, and the pair number may be different in different lines.
+
+For example, the text line `0 1:2 3:4 5:6` in a LibSVM format file indicates that:
+
+- there is a sparse sample `x`, and `x[1]=2`, `x[3]=4`, `x[5]=6`, and the other values of sample `x` are zeros.
+- the label corresponding to the sample is 0.
+
+This kind of file was firstly introduced in [LibSVM](https://www.csie.ntu.edu.tw/~cjlin/libsvm), a library for Support Vector Machines algorithms. After that, this kind of file is widely used in many other machine learning frameworks. For example, [Spark MLLib](https://spark.apache.org/docs/1.0.2/api/python/pyspark.mllib.util.MLUtils-class.html), [XGBoost](https://xgboost.readthedocs.io/en/latest/tutorials/input_format.html) and [SkLearn](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_svmlight_file.html) officially support loading data from this kind of file for training and prediction.
+
+We call this kind of file to be **LibSVM format file** in this document, as all of the machine learning frameworks mentioned above call this file format to be **LibSVM format**.
+
+### LibSVM Format Column
```

Comment:
The table is not exactly the same with LibSVM format. In the example, the label is in a separate column. For LibSVM, the label and KV features are stored in one column in the table.

Reply:
Yes. The standard LibSVM format would contain the label, but it is not required. For example, XGBoost does support to load LibSVM format without the label. In this design, we only support the LibSVM format without the label. Please see [here](https://github.com/sql-machine-learning/sqlflow/pull/2554/files#diff-f73c2f4f93fc028434d604606630b34dR57).

Reply:
So it is not anything like "LibSVM format". We are not supposed to mention LibSVM in this document.